python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -236,44 +237,48 @@ def no_messages_debug_text(self) -> str:
 WAIT_FOR_LOGS_TIMEOUT = 60
 
 
-class PipesBlobStoreMessageReader(PipesMessageReader):
-    """Message reader that reads a sequence of message chunks written by an external process into a
-    blob store such as S3, Azure blob storage, or GCS.
-
-    The reader maintains a counter, starting at 1, that is synchronized with a message writer in
-    some pipes process. The reader starts a thread that periodically attempts to read a chunk
-    indexed by the counter at some location expected to be written by the pipes process. The chunk
-    should be a file with each line corresponding to a JSON-encoded pipes message. When a chunk is
-    successfully read, the messages are processed and the counter is incremented. The
-    :py:class:`PipesBlobStoreMessageWriter` on the other end is expected to similarly increment a
-    counter (starting from 1) on successful write, keeping counters on the read and write end in
-    sync.
-
-    If `log_readers` is passed, the message reader will start the passed log readers when the
-    `opened` message is received from the external process.
-
-    Args:
-        interval (float): interval in seconds between attempts to download a chunk
-        log_readers (Optional[Sequence[PipesLogReader]]): A set of readers for logs.
-    """
-
+class PipesThreadedMessageReader(PipesMessageReader):
     interval: float
     counter: int
-    log_readers: Sequence["PipesLogReader"]
+    log_readers: Mapping[str, "PipesLogReader"]
     opened_payload: Optional[PipesOpenedData]
 
     def __init__(
         self,
         interval: float = 10,
-        log_readers: Optional[Sequence["PipesLogReader"]] = None,
+        log_readers: Optional[
+            Union[Mapping[str, "PipesLogReader"], Sequence["PipesLogReader"]]
+        ] = None,
```

Comment:
would prefer to just canonicalize to one representation or the other

Reply:
This means changing all current implementations including guides and tutorials, also a breaking change. 

Can we do it?

Reply:
I mean this proposes renaming the class right? Seems like we are crossing the boundary of breaking change anyways.

Reply:
Not really renaming. 

99% of the old class functionality is now in a new class. 
The old class now simply inherits from the new class and customizes the remaining 1%. 

What I ended up doing is: 
- update the type annotation 
- but actually still support the old-style list internally so there are no runtime failures